sw/airborne/modules/computer_vision/opticflow_module.c

```diff
@@ -109,27 +128,29 @@ void opticflow_module_run(void)
 {
   pthread_mutex_lock(&opticflow_mutex);
   // Update the stabilization loops on the current calculation
-  if (opticflow_got_result) {
-    uint32_t now_ts = get_sys_time_usec();
-    AbiSendMsgOPTICAL_FLOW(FLOW_OPTICFLOW_ID, now_ts,
-                           opticflow_result.flow_x,
-                           opticflow_result.flow_y,
-                           opticflow_result.flow_der_x,
-                           opticflow_result.flow_der_y,
-                           opticflow_result.noise_measurement,
-                           opticflow_result.div_size);
-    //TODO Find an appropriate quality measure for the noise model in the state filter, for now it is tracked_cnt
-    if (opticflow_result.noise_measurement < 0.8) {
-      AbiSendMsgVELOCITY_ESTIMATE(VEL_OPTICFLOW_ID, now_ts,
-                                  opticflow_result.vel_body.x,
-                                  opticflow_result.vel_body.y,
-                                  0.0f, //opticflow_result.vel_body.z,
-                                  opticflow_result.noise_measurement,
-                                  opticflow_result.noise_measurement,
-                                  -1.0f //opticflow_result.noise_measurement // negative value disables filter updates with OF-based vertical velocity.
-                                 );
+  for (int idx_camera = 0; idx_camera < ACTIVE_CAMERAS; idx_camera++) {
+    if (opticflow_got_result[idx_camera]) {
+      uint32_t now_ts = get_sys_time_usec();
+      AbiSendMsgOPTICAL_FLOW(FLOW_OPTICFLOW_ID, now_ts,
+                             opticflow_result[idx_camera].flow_x,
+                             opticflow_result[idx_camera].flow_y,
+                             opticflow_result[idx_camera].flow_der_x,
+                             opticflow_result[idx_camera].flow_der_y,
+                             opticflow_result[idx_camera].noise_measurement,
+                             opticflow_result[idx_camera].div_size);
+      //TODO Find an appropriate quality measure for the noise model in the state filter, for now it is tracked_cnt
+      if (opticflow_result[idx_camera].noise_measurement < 0.8) {
+        AbiSendMsgVELOCITY_ESTIMATE(VEL_OPTICFLOW_ID, now_ts,
+                                    opticflow_result[idx_camera].vel_body.x,
+                                    opticflow_result[idx_camera].vel_body.y,
+                                    0.0f, //opticflow_result.vel_body.z,
+                                    opticflow_result[idx_camera].noise_measurement,
+                                    opticflow_result[idx_camera].noise_measurement,
+                                    -1.0f //opticflow_result.noise_measurement // negative value disables filter updates with OF-based vertical velocity.
```

Comment:
Similarly, could you add a camera id to the ABI messages? Or use the sender_id for that?

Reply:
There are many files that call these ABI messages, especially for `AbiSendMsgVELOCITY_ESTIMATE()`. Is it okay for me to change all of these to add the extra field? Or would it be better to do this in a separate pull request?

Reply:
Ok, on second thought it might not be necessary for VELOCITY_ESTIMATE, since that is already in body frame anyway.

For the OPTICAL_FLOW message I still think it could be useful since these results are in the camera frame (I think?); and wouldn't you need this for the stop-in-front-of-wall demo anyway? Perhaps a nice way of doing this without too much hassle, is to reserve a short range of sender ids in `subsystems/abi_sender_ids.h` and then calling AbiSendMsgOPTICAL_FLOW with `FLOW_OPTICFLOW_ID + idx_camera` as sender id? That way other modules can directly subscribe to the correct camera, and you won't need to change the function calls anywhere.

Reply:
I don't understand yet how the sender IDs work, for now I've just included the camera_id in the opticflow message, but I will try to implement what you suggested because it might be easier if you can directly subscribe to a single camera instead of having to check for the message you're interested in every time

Reply:
The latest commit should have added what you suggested for the OPTICFLOW message